test/widgets/content_test.dart

```diff
@@ -798,6 +799,19 @@ void main() {
         .single.equals((url: Uri.parse('https://example/'), mode: expectedLaunchMode));
     }, variant: const TargetPlatformVariant({TargetPlatform.android, TargetPlatform.iOS}));
 
+    testWidgets('follow browser preference setting to open URL', (tester) async {
+      await testBinding.globalStore.updateGlobalSettings(
+        eg.globalSettings(
+          browserPreference: BrowserPreference.embedded).toCompanion(false));
+      await prepare(tester,
+        '<p><a href="https://example/">hello</a></p>');
+
+      await tapText(tester, find.text('hello'));
+      check(testBinding.takeLaunchUrlCalls())
+        .single.equals((
+          url: Uri.parse('https://example/'), mode: LaunchMode.platformDefault));
```

Comment:
nit: `.single` compacts nicely onto previous line:
```suggestion
      check(testBinding.takeLaunchUrlCalls()).single.equals((
        url: Uri.parse('https://example/'), mode: LaunchMode.platformDefault));
```